src/CSS.cpp

```diff
@@ -13,36 +13,49 @@ namespace CSS
     {
         sProvider = gtk_css_provider_new();
 
-        struct CSSDir
+        std::vector<std::string> locations;
+        std::string home = std::getenv("HOME");
+        const char* env;
+
+        if ((env = std::getenv("XDG_CONFIG_HOME")))
```

Comment:
```
const char* env = std::getenv("...");
if (env && strlen(env) != 0) {...}
```
Empty environment variable (```strlen(env)==0```) also should use the fallback (Note that ```std::string env = std::getenv("...")``` wouldn't work, because it crashes when given a null c string)

Reply:
The environment variable being empty sounds as likely as the environment variable being set wrong. Should we really test for that?

Reply:
I don't think it is too uncommon to call something like ```XDG_CONFIG_HOME= gBar bar 0``` when you want to temporarily clear a environment variable. Also the xdg base directory spec states that an empty env variable equals fallback.

> I thought a null pointer gets turned into an empty string ... I was being lied to :(

[No, it throws an exception](https://godbolt.org/z/eWadf18xG). Isn't C++ a wonderful language?

Reply:
I have to admit I usually just use something like `unset XDG_CONFIG_HOME` and to me this behavior of `XDG_CONFIG_HOME=` just sounds like some crazy person places their user configs in root. However I have no intention to argue here so I will just add the check 😅